salt/cache/consul.py

```diff
@@ -166,22 +186,37 @@ def list_(bank):
         out = set()
         for key in keys:
             out.add(key[len(bank) + 1 :].rstrip("/"))
-        keys = list(out)
+        keys = [o for o in out if not o.endswith(_tstamp_suffix)]
     return keys
 
 
 def contains(bank, key):
     """
     Checks if the specified bank contains the specified key.
     """
-    if key is None:
-        return True  # any key could be a branch and a leaf at the same time in Consul
-    else:
-        try:
-            c_key = "{}/{}".format(bank, key)
-            _, value = api.kv.get(c_key)
-        except Exception as exc:  # pylint: disable=broad-except
-            raise SaltCacheError(
-                "There was an error getting the key, {}: {}".format(c_key, exc)
-            )
-        return value is not None
+    key = key or ""
+    try:
+        c_key = "{}/{}".format(bank, key)
+        _, value = api.kv.get(c_key, keys=True)
+    except Exception as exc:  # pylint: disable=broad-except
+        raise SaltCacheError(
+            "There was an error getting the key, {}: {}".format(c_key, exc)
+        )
+    return value is not None
```

Comment:
```suggestion
    try:
        c_key = "{}/{}".format(bank, key or "")
        _, value = api.kv.get(c_key, keys=True)
    except Exception as exc:  # pylint: disable=broad-except
        raise SaltCacheError(
            "There was an error getting the key, {}: {}".format(c_key, exc)
        )
    return value is not None
```
This suggestion is just personal taste, since we don't use the variable anywhere else in the function body

Reply:
done